qiskit/circuit/classical/types/types.py

```diff
@@ -89,6 +89,9 @@ class Bool(Type, metaclass=_Singleton):
     def __repr__(self):
         return "Bool()"
 
+    def __hash__(self):
+        return hash(self.__class__)
```

Comment:
Honestly I don't really know why I went for one or the other.  I can use `type` if you prefer - I don't think there's much difference for Python-space types (but could be forgetting something).

Reply:
Yeah, this doesn't matter and it's fine to keep it like this. I was just wondering if there was an advantage one way or the other because my instinct would be to use `type(self)`.